src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/reissuefurtherevidence/ReissueFurtherEvidenceAboutToStartHandler.java

```diff
@@ -77,23 +79,30 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
 
     private void setDocumentDropdown(SscsCaseData sscsCaseData, List<? extends AbstractDocument> availableDocumentsToReIssue) {
         List<DynamicListItem> listCostOptions = new ArrayList<>();
+        boolean isConfidentialCase = isYes(sscsCaseData.getIsConfidentialCase());
 
         for (AbstractDocument doc : availableDocumentsToReIssue) {
-            String label = buildFormattedLabel(doc);
+            String label = buildFormattedLabel(doc, isConfidentialCase);
             if (doc.getValue().getDocumentLink() != null) {
-                listCostOptions.add(new DynamicListItem(doc.getValue().getDocumentLink().getDocumentUrl(), label));
+                DocumentLink documentLink = isConfidentialCase ? ofNullable(doc.getValue().getEditedDocumentLink())
+                    .orElse(doc.getValue().getDocumentLink()) : doc.getValue().getDocumentLink();
+                listCostOptions.add(new DynamicListItem(documentLink.getDocumentUrl(), label));
             }
         }
 
         sscsCaseData.getReissueArtifactUi().setReissueFurtherEvidenceDocument(new DynamicList(listCostOptions.get(0), listCostOptions));
     }
 
-    private String buildFormattedLabel(AbstractDocument doc) {
-        String filenameLabel = doc.getValue().getDocumentFileName();
-        if (doc instanceof SscsWelshDocument) {
-            filenameLabel = getBilingualLabel(doc);
+    private String buildFormattedLabel(AbstractDocument doc, boolean isConfidentialCase) {
+        if (isConfidentialCase && doc.getValue().getEditedDocumentLink() != null) {
+            return doc.getValue().getEditedDocumentLink().getDocumentFilename();
+        } else {
```

Comment:
does this logic not apply if the document is confidential?

Reply:
It shouldn't. When evidence share checks if it should send the original or the edited version, it checks if the case is confidential first.